prov/sockets/src/sock_ep.c

```diff
@@ -1417,7 +1417,8 @@ static void sock_set_fabric_attr(void *src_addr, const struct fi_fabric_attr *hi
 	if (!attr->name)
 		attr->name = strdup(sock_fab_name);
 
-	attr->prov_name = NULL;
+	attr->prov_name = hint_attr->prov_name;
```

Comment:
This needs to be strdup.  Otherwise it's pointing the prov_name to the same string that hint_attr points to.  If hint_attr are freed, attr->prov_name will reference invalid memory.